instrumentation/apache-httpclient/apache-httpclient-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/v5_0/ApacheHttpAsyncClientInstrumentation.java

```diff
@@ -264,6 +264,9 @@ private void cancelDelegate() {
 
     @Nullable
     private HttpResponse getResponseFromHttpContext() {
+      if (httpContext instanceof HttpCoreContext) {
+        return ((HttpCoreContext) httpContext).getResponse();
```

Comment:
`HttpCoreContext.HTTP_RESPONSE` seems to be deprecated in later versions and using the getter is now preferred. In opensearch3 tests `httpContext.getAttribute(HttpCoreContext.HTTP_RESPONSE)` returns `null`.